packages/http/src/validator/__tests__/index.spec.ts

```diff
@@ -71,80 +73,44 @@ describe('HttpValidator', () => {
                 request: { body: { contents: [], required: true } },
                 responses: [{ code: '200' }],
               },
-              4,
-            ),
+              undefined,
+              1
+            )
           );
         });
       });
     });
 
     describe('headers validation in enabled', () => {
-      const validate = (resourceExtension?: Partial<IHttpOperation>, length = 1) => () => {
-        expect(
-          validateInput({
-            resource: Object.assign(
-              {
-                method: 'get',
-                path: '/',
-                id: '1',
-                request: {},
-                responses: [{ code: '200' }],
-              },
-              resourceExtension,
-            ),
-            element: { method: 'get', url: { path: '/' } },
-          }),
-        ).toHaveLength(length);
-      };
-
       describe('request is not set', () => {
-        it('validates headers', validate(undefined, 3));
+        it('does not validate headers', validate(undefined, undefined, 0));
```

Comment:
there's a couple of these, does it mean that there's a user-facing change in Prism behavior or is it just a part of the refactor?

Reply:
Good question. So previously our validators were ran even if there was no need to run them. You can see it here: https://github.com/stoplightio/prism/pull/731/files#diff-52fafd78c752aca126e4fd8c4ce567a6L45-L47

You can see that the functions were called anyway even if there was nothing to validate. In the following lines instead you can see there's a check to make sure it's the case to call them or otherwise call directly `Either.right`

In this tests, the validators were mocked to return an error in any case, so no matter what was happening, all the validators were always called and returning an error. Body + Query + Headers = 3 validators — that explains the number.